osu.Game/Screens/Edit/Screens/Setup/Components/SetupTickSliderBar.cs

```diff
@@ -0,0 +1,513 @@
+                                new Container
+                                {
+                                    RelativePositionAxes = Axes.X,
+                                    Anchor = Anchor.Centre,
+                                    Origin = Anchor.Centre,
+                                    Child = middleTickCaption = new OsuSpriteText
+                                    {
+                                        Anchor = Anchor.TopCentre,
+                                        Origin = Anchor.TopCentre,
+                                        Alpha = 0.8f,
+                                        TextSize = default_caption_text_size,
+                                    }
+                                },
+                                new Container
+                                {
+                                    RelativePositionAxes = Axes.X,
+                                    Anchor = Anchor.CentreRight,
+                                    Origin = Anchor.CentreRight,
+                                    Child = rightTickCaption = new OsuSpriteText
+                                    {
+                                        Anchor = Anchor.TopRight,
+                                        Origin = Anchor.TopRight,
+                                        Alpha = 0.8f,
+                                        TextSize = default_caption_text_size,
+                                    }
+                                }
+                            },
+                        }
+                    }
+                },
+                new HoverClickSounds()
+            };
+
+            Nub.Current.Value = true;
+
+            ticks.TickClicked += a => Current.Value = a;
+        }
+
+        [BackgroundDependencyLoader]
+        private void load(AudioManager audio, OsuColour colours)
+        {
+            sample = audio.Sample.Get(@"UI/sliderbar-notch");
+            AccentColour = colours.BlueDark;
+            Nub.Colour = colours.BlueDark;
+            Nub.AccentColour = colours.BlueDark;
+            Nub.GlowColour = colours.BlueDarker;
+            Nub.GlowingAccentColour = colours.BlueLighter;
+            leftTickCaption.Colour = colours.BlueDark;
+            middleTickCaption.Colour = colours.BlueDark;
+            rightTickCaption.Colour = colours.BlueDark;
+        }
+
+        protected override bool OnHover(InputState state)
+        {
+            Nub.Glowing = true;
+            return base.OnHover(state);
+        }
+
+        protected override void OnHoverLost(InputState state)
+        {
+            Nub.Glowing = false;
+            base.OnHoverLost(state);
+        }
+
+        protected override bool OnKeyDown(InputState state, KeyDownEventArgs args)
+        {
+            leftShiftHeld |= args.Key == Key.ShiftLeft;
+            rightShiftHeld |= args.Key == Key.ShiftRight;
+            IsUsingAlternatePrecision = leftShiftHeld | rightShiftHeld;
```

Comment:
`state.Keyboard.ShiftPressed`

Reply:
I had encountered some issues regarding this. The reason why I implemented it like that was the conflict that was caused when both Shifts were pressed or held simultaneously. Since I haven't followed up with the changes, I don't know if this can be safely removed or not.

Reply:
It's been known that holding both shifts and releasing them, the state will be bugged. The first shift that was pressed will have its status correctly updated on release, however the second one will appear held. Pressing and releasing the second shift again will then fix the state.